scripts/check_cpp_format.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+
+import fnmatch
+import io
+import multiprocessing
+import os
+import queue
+import subprocess
+import sys
+import threading
+
+# Parse command-line arguments
+if len(sys.argv) == 1:
+    print("Usage: check_cpp_format.py [-fix] <paths>")
+    sys.exit()
+i_arg = 1
+fix = False
+if sys.argv[i_arg] == "-fix":
+    fix = True
+    i_arg += 1
+roots = []
+for root in sys.argv[i_arg:]:
+    roots.append(root)
+
+# Parse environment variables
+prune_paths = set()
+if "PRUNE_PATHS" in os.environ:
+    last_token = io.StringIO()
+    in_quotes = False
+    for ch in os.environ["PRUNE_PATHS"]:
+        if ch in ["'", '"']:
+            if last_token.tell() != 0:
+                prune_paths.add(last_token.getvalue())
+                last_token = io.StringIO()
+            in_quotes = not in_quotes
+        elif ch == " ":
+            if in_quotes:
+                last_token.write(ch)
+            elif last_token.tell() != 0:
+                prune_paths.add(last_token.getvalue())
+                last_token = io.StringIO()
+            # else tokens are separated by multiple spaces
+        else:
+            last_token.write(ch)
+    if last_token.tell() != 0:
+        prune_paths.add(last_token.getvalue())
+        last_token = io.StringIO()
```

Comment:
Please don't write a parser manually. Regexs are better than this even if they are a pain in their own right. And I think you could rewrite this using `str.split` and a few other things if you really don't want to use regexs. I cannot tell what's happening here so it needs to be simplified.

Reply:
`str.split` definitely fails, because we get on input the paths in quotes, and the paths may contain spaces. Cryptic regex would not be easier to read than the parser code.